types/storybook__react/index.d.ts

```diff
@@ -12,15 +13,21 @@ import * as React from 'react';
 export type Renderable = React.ComponentType | JSX.Element;
 export type RenderFunction = () => Renderable | Renderable[];
 
+export interface DecoratorParameters {
+    [key: string]: any;
+}
 export type StoryDecorator = (story: RenderFunction, context: { kind: string, story: string }) => Renderable | null;
 
 export interface Story {
     readonly kind: string;
-    add(storyName: string, callback: RenderFunction): this;
+    add(storyName: string, callback: RenderFunction, parameters?: DecoratorParameters): this;
```

Comment:
Here's where this method is defined: https://github.com/storybooks/storybook/blob/next/lib/core/src/client/preview/client_api.js#L92

And this line shows that `parameters` is optional: https://github.com/storybooks/storybook/blob/next/lib/core/src/client/preview/client_api.js#L111